sysdump/sysdump.go

```diff
@@ -793,6 +833,32 @@ func (c *Collector) submitBugtoolTasks(ctx context.Context, pods []*corev1.Pod,
 	return nil
 }
 
+func (c *Collector) submitHubbleFlowsTasks(_ context.Context, pods []*corev1.Pod, containerName string, path func(string) string) error {
+	for _, p := range pods {
+		p := p
+		if err := c.pool.Submit(fmt.Sprintf("hubble-flows-"+p.Name), func(ctx context.Context) error {
+			// HACK: Run hubble observe with --follow to avoid hitting the code path that triggers
+			// https://github.com/cilium/cilium/issues/17036.
+			b, e, err := c.client.ExecInPodWithStderr(ctx, p.Namespace, p.Name, containerName, []string{
+				"timeout", "--signal", "SIGINT", "--preserve-status", "5", "bash", "-c",
+				fmt.Sprintf("hubble observe --follow --last %d -o jsonpb", c.options.HubbleFlowsCount),
```

Comment:
I am a bit surprised how `--follow` helps with https://github.com/cilium/cilium/issues/17036 - because the code that can hit the overwritten data is executed in both cases.

Another approach we could try is to limit `HubbleFlowsCount` to not dump the whole ring buffer, but only the last 75% or so. This makes it much more unlikely that we ever cross paths with the writer.

Reply:
Discussed offline - I was wrong. Follow-mode does indeed not trigger the error.